Tests/CommunicationLayer/CommDeviceControl/UdpMessageForwarderTest.cpp

```diff
@@ -0,0 +1,203 @@
+/**
+ *  Schulich Delta Hermes
+ *  Copyright (C) 2015 University of Calgary Solar Car Team
+ *
+ *  This file is part of Schulich Delta Hermes
+ *
+ *  Schulich Delta Hermes is free software:
+ *  you can redistribute it and/or modify it under the terms
+ *  of the GNU Affero General Public License as published by
+ *  the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  Schulich Delta Hermes is distributed
+ *  in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or
+ *  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+ *  General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General
+ *  Public License along with Schulich Delta Hermes.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  For further contact, email <software@calgarysolarcar.ca>
+ */
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <QScopedPointer>
+#include <QDebug>
+#include <QByteArray>
+#include <string>
+
+#include <SimpleAmqpClient/SimpleAmqpClient.h>
+
+#include "CommunicationLayer/CommDeviceControl/UdpMessageForwarder.h"
+#include "InfrastructureLayer/Settings/MockSettings.h"
+
+using ::testing::Return;
+
+namespace
+{
+    const QString MOCK_IP = QString("localhost");
+    const quint16 MOCK_PORT = 5672;
+    const QString MOCK_QUEUE = "test-queue";
+    const QString MOCK_EXCHANGE = "amq.direct";
+    const QString MOCK_ROUTING_KEY = "routing-key";
+    const QString EXPECTED_1 = "Message Test";
+    const QString EXPECTED_2 = "Second Message Test";
+    const QString EXPECTED_3 = "Sp3c1a1** Ch4r4c74r5__ 7357";
+}
+
+class UdpMessageForwarderTest : public ::testing::Test
+{
+protected:
+
+    QScopedPointer<MockSettings> settings_;
+    QScopedPointer<UdpMessageForwarder> forwarder;
+    AmqpClient::Channel::ptr_t receiver;
+
+    // TODO test that disconnecting the sender does not destroy messages in the queue
+
+    /**
+   * @brief SetUp will set up the receiver to verify messages are being sent, as well as mocking the settings to be used by the UdpMessageForwarder
```

Comment:
Hey I thought this wasn't really testable, but creating a new receiver to take in the messages is great!